### ISSUE TITLE
[Bug Report] AllocationPolicy field for enrollment group set to "null" when retrieving the group or mapping the creation/update response

### ISSUE THREAD

**Issue by dusancvetkov** [Azure/azure-iot-sdk-java#1286]
# Context

- **OS and version used:** Windows 10 
- **Java runtime used:** openjdk version "11.0.9.1" 2020-11-04
- **SDK version used:** provisioning-service-client 1.7.0 & 1.8.0 (both tested)

## Description of the issue

Java IoT SDK could not map the api response properly when creating/updating/retrieving an enrollment group.
The field name that could not be mapped is "allocationPolicy", if we provide the "CUSTOM" (custom allocation policy) value, it will be saved to the enrollment group in the azure dps service, but when trying to retrieve it the value is just null. 

## Code sample exhibiting the issue

```
EnrollementGroup enrollmentGroup = new EnrollmentGroup(....);
enrollmentGroup.setAllocationPolicy(AllocationPolicy.CUSTOM);

EnrollementGroup resultAfterCreation = provisioningServiceClient.createOrUpdateEnrollmentGroup(enrollmentGroup);
log.info(resultAfterCreation.getAllocationPolicy()); // NULL but allocation policy is properly set in DPS

// if we try to update the group with this "null" value, the default provisioning will be set 
EnrollementGroup resultAfterUpdate = provisioningServiceClient.createOrUpdateEnrollmentGroup(resultAfterCreation);
log.info(resultAfterUpdate.getAllocationPolicy()); // NULL again & now allocation policy in dps for this group is set to default

```

My assumption is that the problem occurs in class: 

**com.microsoft.azure.sdk.iot.provisioning.service.configs.EnrollmentGroup** line **126** when the json string response could not map the enum field **com.microsoft.azure.sdk.iot.provisioning.service.configs.AllocationPolicy** because it returns it with value "custom" and not with value "CUSTOM" or could not map the enum value at all. 

## Console log of the issue

N/A

**Comment by timtay-microsoft** [comment]
Sounds like a bug, yeah. We'll get this fixed ASAP

**Comment by timtay-microsoft** [comment]
#1289 will fix this issue for you.

**Comment by dusancvetkov** [comment]
Thanks @timtay-microsoft for the fast response. When should we expect the release with this bugfix to be pushed?

**Comment by timtay-microsoft** [comment]
I'll try to get a release out with this fix within the next few work days.

**Comment by timtay-microsoft** [comment]
Closing this issue as the newly released version provisioning service client 1.9.0 contains the fix for this issue

**Comment by dusancvetkov** [comment]
Thanks a lot!
